plugins/modules/system/listen_ports_facts.py

```diff
@@ -181,10 +195,87 @@ def netStatParse(raw):
     return results
 
 
-def main():
+def ss_parse(raw):
+    results = list()
+    regex_conns = re.compile(pattern=r'\[?(.+?)\]?:([0-9]+)')
+    regex_pid = re.compile(pattern=r'"(.*?)",pid=(\d+)')
+
+    lines = raw.splitlines()
+
+    if len(lines) == 0 or not lines[0].startswith('Netid '):
+        # unexpected stdout from ss
+        raise EnvironmentError('Unknown stdout format of `ss`: {0}'.format(raw))
+
+    # skip headers (-H arg is not present on e.g. Ubuntu 16)
+    lines = lines[1:]
+
+    for line in lines:
+        cells = line.split()
```

Comment:
Is it possible that `process` contains spaces? In that case, having a value for `maxsplit` would be a good idea:
```suggestion
        cells = line.split(None, 6)
```

Reply:
I would be very surprised if that's a real thing, but not sure if OS'es forbid it in any way.

Reply:
I've never seen a process name with spaces or any whitespace char on any linux distro, so I would also be very surprised.
But having maxsplit does not harm anything, so I'll integrate it.